teps/0066-dogfooding-tekton.md

```diff
@@ -0,0 +1,320 @@
+    a problems related to maintainability and scale
+- Offer CI/CD services to the Tekton community with minimal dependencies
+  to other products, and thus greater control on the outcome
+- Build a repository of examples that embody best practices for Tekton use to tap in
+
+## Requirements
+
+- Provide reliable CI/CD services to the Tekton community
+- Dogfood as many project projects as possible, from the most stable and mature to
+  the experimental ones
+- Provide a mean to experiment with alpha and experimental features and build a way
+  forward to stable
+
+## Proposal
+
+<!--
+This is where we get down to the specifics of what the proposal actually is.
+This should have enough detail that reviewers can understand exactly what
+you're proposing, but should not include things like API designs or
+implementation.  The "Design Details" section below is for the real
+nitty-gritty.
+-->
+
+### Notes/Caveats (optional)
+
+<!--
+What are the caveats to the proposal?
+What are some important details that didn't come across above.
+Go in to as much detail as necessary here.
+This might be a good place to talk about core concepts and how they relate.
+-->
+
+### Risks and Mitigations
+
+<!--
+What are the risks of this proposal and how do we mitigate. Think broadly.
+For example, consider both security and how this will impact the larger
+kubernetes ecosystem.
+
+How will security be reviewed and by whom?
+
+How will UX be reviewed and by whom?
+
+Consider including folks that also work outside the WGs or subproject.
+-->
+
+We want to offer continuous integration and continuous delivery services to
+the Tekton community, which are essential to the development of Tekton itself.
+This goes in contrast with the need of dogfooding alpha stability and even
+experimental services. Mitigations for this are:
+
+- Use a multi-cluster setup:
+  - One cluster runs nightly releases and experimental code. This cluster may
+    offer services, but no essential ones, so that Tekton development may continue
+    normally if this cluster is broken.
+  - One cluster runs major releases of Tekton components (when available).
+    Deployments to this cluster are vetted by humans. Write / admin access to this
+    cluster is reduced to a limited number of individuals. This cluster may offer
+    essential services.
+- The build-captain role: there is always at least on person per day on duty to
+  verify the status of CI/CD services and respond to incidents.
+- Introduce Tekton based services incrementally. For experimental components, if
+  an alternative is available, use the experimental component only in a few places
+  until the component has proven stable enough.
+
+Using experimental Tekton components in production may raise security concerns.
+Mitigations are somewhat similar to those already discussed:
+
+- Use experimental components for non essential services. This allows stopping
```

Comment:
would this mean we couldn't use the task loops experimental custom task in our dogfooding cluster?

Reply:
That's a good point. I think we could still use it, but we just need to be aware that there's a risk involved.
Perhaps documenting these risks and mitigations in the `build-captain` runbook would help?

Reply:
done